src/main/scala/fi/oph/koski/validation/KoskiValidator.scala

```diff
@@ -1124,4 +1125,22 @@ class KoskiValidator(
       )(KoskiErrorCategory.badRequest.validation.rakenne.deprekoituOsaamisenHankkimistapa())
     case _ => HttpStatus.ok
   }
+
+  private def validateLukioonValmistava2019(suoritus: Suoritus) = {
+    suoritus match {
+      case s: LukioonValmistavanKoulutuksenSuoritus => validateLukioonValmistava2019Osasuoritukset(s)
+      case _ => HttpStatus.ok
+    }
+  }
+
+  private def validateLukioonValmistava2019Osasuoritukset(suoritus: LukioonValmistavanKoulutuksenSuoritus) = {
+    if (suoritus.osasuoritukset.getOrElse(List()).exists(_.isInstanceOf[LukionOppiaineenOpintojenSuoritusLukioonValmistavassaKoulutuksessa2019]) &&
+      suoritus.osasuoritukset.getOrElse(List()).exists(_.isInstanceOf[LukionOppiaineenOpintojenSuoritusLukioonValmistavassaKoulutuksessa]))
+    {
+      KoskiErrorCategory.badRequest.validation.rakenne.lukioonValmistavassaEriLukioOpsienOsasuorituksia()
+    }
+    else {
+      HttpStatus.ok
+    }
+  }
```

Comment:
Lukio2019-validointeja tehdessä oli muistaakseni puhetta, että alettaisiin muutenkin tästä eteenpäin tekemään näitä spesifejä validointeja erillisiin objekteihin/luokkiin, jotta KoskiValidator ei paisuisi enää enempää. Nääkin voisi samaan tyyliin siirtää omaan tiedostoonsa?